### DISCUSSION TITLE
How to use Snap with Angular

### DISCUSSION THREAD

**Discussion by yaniv-fink** [MetaMask/snaps#1129] [Q&A] [answered]
The snap React example project is divided into a snap folder and a site folder both inside a packages folder

I want to add the snap to an existing **Angular project** (and not build it from scratch with react)
The site folder is no problem - since it just calls the wallet functionality, 
but the question is how to import the Snap folder (from the react example) into the Angular project, should I use npm install or another way?
I am sure the developers did it generic so you can just use the snap RPC somehow, and did not mean to create it only for React.

Right now when I am calling wallet_enable from the website, it doesn't have any information about the snap folder, and that's why 
the result of the returning object looks something like this, which is reasonable.
![image](https://user-images.githubusercontent.com/9554134/212687194-cd4c580a-a5f6-4eab-aae3-543c8b8d0618.png)

So again, the question is how exactly to tell my Angular project to be familiar with the Snap folder, and is there any example I can learn from?
maybe Snap with other languages

Thank you.

**Comment by ritave** [comment, marked as answer]
Hey @yaniv-fink, we also have more bare-bones templates outside of React

1. [Typescript template](https://github.com/MetaMask/template-typescript-snap)
2. [JavaScript template](https://github.com/MetaMask/template-snap)

I believe the CLI also can set-up the TypeScript one for you - `mm-snaps init`. We don't have the scripts set-up for Angular, so the Snaps CLI will be your main point of setting up the snap and testing it.

In general, you should treat a snap as a separate Node package from the Angular project, we used the monorepo pattern for the React template for easier quickstart, but it would also work if the react website was in one repository and the snap in another.

You can develop your snap normally using localhost snaps and the CLI's `mm-snaps serve` command, after you're ready to publish to your users, you can publish the snap to the NPM registry and use the new `npm:my-snap` ID instead.

**Reply by yaniv-fink** [reply]
Thank you!
I like the idea of a separate project  (package),  this way I can encapsulate other functionality like maybe
connectSnap, getSnap, isFlask and the RPC, all will be in this package
I think the community will benefit from such an example in your documentation. (configurable port and so on)

**Reply by Montoya** [reply]
Something cool that you can try is to build your snap in template-snap-monorepo and run it from the CLI so the snap is available on port 8080. Then in your angular project you can reference this snap at this address (localhost:8080). Since everything is running locally on your computer you can test installing the snap this way. 

Later when you publish your snap to NPM, you can reference it from your angular project with the npm project ID instead of localhost.

**Reply by yaniv-fink** [reply]
Very cool thank you @Montoya for this interesting solution!
